osi_logicallane.proto

```diff
@@ -0,0 +1,502 @@
+    // All LogicalLaneBoundary elements referenced here must refer to the same
+    // ReferenceLine as this lane.
+    // The boundaries together must cover the whole length of the lane (the
+    // range [start_s,end_s]) without gap or overlap. The boundaries must be
+    // stored in ascending order, starting with the smallest S position.
+    // Consecutive boundaries must share a point: the last point of the
+    // previous boundary must be identical to the first point of the next
+    // boundary.
+    //
+    // TODO Is it allowed for the boundaries to be longer than the lane?
+    // Theoretically, this could be done. Then the user would have to limit the
+    // boundary itself to [start_s,end_s].
+    //
+    repeated Identifier right_boundary_id = 12;
+
+    // Left boundary of this lane.
+    // References to LogicalLaneBoundary elements.
+    // All LogicalLaneBoundary elements referenced here must refer to the same
+    // ReferenceLine as this lane.
+    // The boundaries together must cover the whole length of the lane (the
+    // range [start_s,end_s]) without gap or overlap. The boundaries must be
+    // stored in ascending order, starting with the smallest S position.
+    // Consecutive boundaries must share a point: the last point of the
+    // previous boundary must be identical to the first point of the next
+    // boundary.
+    //
+    // TODO Is it allowed for the boundaries to be longer than the lane?
+    // Theoretically, this could be done. Then the user would have to limit the
+    // boundary itself to [start_s,end_s].
+    //
+    repeated Identifier left_boundary_id = 13;
+
+    // Lanes that directly are connected to this lane at the beginning
+    //
+    repeated LaneConnection predecessor_lane = 14;
+
+    // Lanes that directly are connected to this lane at the end
+    //
+    repeated LaneConnection successor_lane = 15;
+
+    // Connection to another lane (predecessor or successor)
+    //
+    message LaneConnection {
+
+        // Id of the other logical lane that is connected
+        //
+        optional Identifier other_lane_id = 1;
+
+        // If true: LaneConnection is at the beginning of the other lane
+        // If false: LaneConnection is a the end of the other lane
+        //
+        optional bool at_begin_of_other_lane = 2;
+
+    }
+
+    // Relation of this lane to another logical lane
+    //
+    message LaneRelation {
+
+        // Id of the other logical lane
+        //
+        optional Identifier other_lane_id = 1;
+
+        // Start s position of the relation
+        //
+        optional double start_s = 2;
+
+        // End s position of the relation
+        //
+        // Requirement: end_s > start_s
+        //
+        optional double end_s = 3;
+
+        // Start s position of the relation on the other lane.
+        //
+        // This is the same place as start_s, but measured along the reference
+        // line of the other lane.
```

Comment:
In line 362 you recommend for neighboring lanes to use the same reference line. So when adopting this recommendation this field here would have the same value as start_s?

Reply:
I'll have to clarify this one. I'll add some additional text and pictures later.

Basically one can use the same reference line if the lanes are parallel for their whole length.
But if you think of an intersection with a left-turn and a right-turn lane, e.g like this:
```
        /     /
-------/     /
left turn   /
-----------X
right turn  \
---------\   \
          \   \
```
These are neighbours for a while (which should be reflected in their lane relations), but cannot reasonably share a reference line. Since the reference line might even be in different directions for the area where they are neighbours (can be done in OpenDRIVE), their s values might be hugely different. That's why separate values per lane are needed there.

Reply:
Added some explanation.